src/main/java/com/dumbdogdiner/stickyapi/common/util/StringUtil.java

```diff
@@ -287,4 +307,58 @@ public static UUID hyphenateUUID(@NotNull String uuid) {
             return UUID.fromString(uuid);
         }
     }
+
+
+    /**
+     * Creates a random string of "magic" characters of characters
+     *
+     * @param min               the minimum length of the string (inclusive)
+     * @param max               the maximum length of the string (inclusive)
+     * @param minRunBeforeSpace Minimum number of characters before a space can appear, set to 0 to disable space
+     * @return A string of random characters, where the length
+     */
+    public static String randomObfuscatedString(int min, int max, int minRunBeforeSpace) {
+        Preconditions.checkArgument(max >= min, "Max cannot be less than min");
+        // Placeholder chars by width
+        char[] choices = new char[]{
+                // Space is boring
+                // ' ', //  1 px
+                'i', //  2 px
+                'l', //  3 px
+                't', //  4 px
+                'f', //  5 px
+                'a', //  6 px
+                '@'  //  7 px
+        };
+        StringBuilder obfuscated = new StringBuilder();
+
+        int len = MathUtil.randomInt(min, max);
+        int charsSinceSpace = 0;
+        while (obfuscated.length() < len) {
+            if (    minRunBeforeSpace > 0 &&
+                    charsSinceSpace > minRunBeforeSpace &&
+                    // Set a 5% probability of the character being a space
+                    MathUtil.randomInt(1, 100) <= 5) {
+                obfuscated.append(' ');
+                charsSinceSpace = 0;
+            } else {
+                obfuscated.append(MathUtil.randomElement(choices));
+                charsSinceSpace++;
+            }
+        }
+        obfuscated.insert(0, ChatColor.MAGIC);
+        obfuscated.append(ChatColor.RESET);
+        return obfuscated.toString();
+    }
+
+    /**
+     * Replaces &amp; followed by any valid minecraft format code (matching the regex <pre>(?=([a-f]|[0-9]|[klmnor]))</pre> with &#x00a7;
+     *
+     * @param input The input string
+     * @return A string where the relevant ampersands are replaced with section symbols
+     */
+    public static String formatChatCodes(String input) {
```

Comment:
There's a similar function in the Translation class, would be worth updating it to use this. Although iirc @SkyezerFox was going to rewrite translation...